game/scripts/vscripts/abilities/oaa_bristleback.lua

```diff
@@ -0,0 +1,108 @@
+    if result_angle >= (180 - (ability:GetSpecialValueFor("back_angle") / 2)) and result_angle <= (180 + (ability:GetSpecialValueFor("back_angle") / 2)) then
+      -- Check if the reduced damage is lethal
+      if((params.unit:GetHealth() - (params.Damage * (1 - back_reduction_percentage))) >= 1) then
+        -- This is the actual "damage reduction".
+        params.unit:SetHealth((params.Damage * back_reduction_percentage) + params.unit:GetHealth())
+        -- Play the sound on Bristleback.
+        EmitSoundOn(params.sound, params.unit)
+        -- Create the back particle effect.
+        local back_damage_particle = ParticleManager:CreateParticle(params.back_particle, PATTACH_ABSORIGIN_FOLLOW, params.unit)
+        -- Set Control Point 1 for the back damage particle; this controls where it's positioned in the world. In this case, it should be positioned on Bristleback.
+        ParticleManager:SetParticleControlEnt(back_damage_particle, 1, params.unit, PATTACH_POINT_FOLLOW, "attach_hitloc", params.unit:GetAbsOrigin(), true)
+        -- Increase the Quill Spray damage counter based on how much damage was done *post-Bristleback mitigation*.
+        params.unit.quill_threshold_counter = params.unit.quill_threshold_counter + (params.Damage - (params.Damage * back_reduction_percentage))
+      end
+    else
+      -- Check if the reduced damage is lethal
+      if (params.unit:GetHealth() - (params.Damage * (1 - side_reduction_percentage))) >= 1 then
+        -- This is the actual "damage reduction".
+        params.unit:SetHealth((params.Damage * side_reduction_percentage) + params.unit:GetHealth())
+        -- Play the sound on Bristleback.
+        EmitSoundOn(params.sound, params.unit)
+        -- Create the side particle effect.
+        local side_damage_particle = ParticleManager:CreateParticle(params.side_particle, PATTACH_ABSORIGIN_FOLLOW, params.unit)
+        -- Set Control Point 1 for the side damage particle; same stuff as the back damage particle.
+        ParticleManager:SetParticleControlEnt(side_damage_particle, 1, params.unit, PATTACH_ABSORIGIN_FOLLOW, "attach_hitloc", params.unit:GetAbsOrigin(), true)
+        ParticleManager:SetParticleControlEnt(side_damage_particle, 2, params.unit, PATTACH_ABSORIGIN_FOLLOW, "attach_hitloc", Vector(0, result_angle, 0), true)
+        -- Increase the Quill Spray damage counter based on how much damage was done *post-Bristleback mitigation*.
+        params.unit.quill_threshold_counter = params.unit.quill_threshold_counter + (params.Damage - (params.Damage * side_reduction_percentage))
+      end
+    end
+  end
+
+  local quill_spray = params.unit:FindAbilityByName("bristleback_quill_spray")
+
+  -- If the amount of damage taken since the last Quill Spray proc is equal to or exceeds what's defined as the threshold, release a Quill Spray.
+  if params.unit.quill_threshold_counter >= ability:GetSpecialValueFor("quill_release_threshold") and ability:IsCooldownReady() then
```

Comment:
Not really. That one was an issue of how the stored damage reset. Even vanilla behaves like set to zero I'm pretty sure, so I'm not sure why it was like that in the first place.
What happens with that code would be that if Bristle took 5 times the threshold damage in the back, he would trigger quills for any damage the next 4 times it was off cooldown, in addition to the initial proc.